rust/src/smb/smb.rs

```diff
@@ -682,22 +693,31 @@ pub fn u32_as_bytes(i: u32) -> [u8;4] {
     return [o1, o2, o3, o4]
 }
 
-#[derive(Default, Debug)]
+#[derive(Debug)]
 pub struct SMBState<> {
     pub state_data: AppLayerStateData,
 
     /// map ssn/tree/msgid to vec (guid/name/share)
-    pub ssn2vec_map: HashMap<SMBCommonHdr, Vec<u8>>,
+    pub ssn2vec_cache: LruCache<SMBCommonHdr, Vec<u8>>,
```

Comment:
Next question after this work :
So, this work bounds the number of elements by flow (great), but is the size of each element bounded ? Can this Vec<u8> and other ones like `guid2name_cache: LruCache<Vec<u8>, Vec<u8>>` have arbitrary big vectors ?